### PR TITLE
Added mob spawning!

### PR DESCRIPTION
**Pull request by matcracker**
Added all mobs spawning without AI

### CODE REVIEW

**Comment by matcracker** [comment]
Don't merge yet

File: src/pocketmine/block/Pumpkin.php [hunk 1]
```diff
@@ -56,6 +62,76 @@ public function place(Item $item, Block $block, Block $target, $face, $fx, $fy,
...
 		$this->getLevel()->setBlock($block, $this, true, true);
 
+		if($player != null){
+			$firstBlock = $this->getLevel()->getBlock($block->add(0, -1, 0));
+			$secondBlock = $this->getLevel()->getBlock($block->add(0, -2, 0));
+			$thirdBlock = $this->getLevel()->getBlock($block->add(-1, -1, 0));
+			$fourthBlock = $this->getLevel()->getBlock($block->add(1, -1, 0));
+			$fifthBlock = $this->getLevel()->getBlock($block->add(0, -1, -1));
+			$sixthBlock = $this->getLevel()->getBlock($block->add(0, -2, 1));
+
```
Comment (thebigsmileXD) [inline src/pocketmine/block/Pumpkin.php]: Using this method works, but isn't standard. $this->getSide() is more relieable

File: src/pocketmine/entity/Bat.php [hunk 2]
```diff
@@ -56,26 +58,11 @@ public function getName(){
 	}
 
 	 public function spawnTo(Player $player){
-		$pk = new AddEntityPacket();
-		$pk->eid = $this->getId();
+		$pk = $this->addEntityDataPacket($player);
```
Comment (thebigsmileXD) [inline src/pocketmine/entity/Bat.php]: Nice function. Removes all the mess

File: src/pocketmine/entity/Mooshroom.php [hunk 3]
```diff
@@ -26,38 +26,45 @@
...
+			drp::get(drp::LEATHER, 0, mt_rand(0, 2))
+		];
+
+		if($this->getLastDamageCause() === EntityDamageEvent::CAUSE_FIRE){
+			$drops[] = drp::get(drp::COOKED_BEEF, 0, mt_rand(1, 3));
+		}else{
+			$drops[] = drp::get(drp::RAW_BEEF, 0, mt_rand(1, 3));
+		}
+		//TODO Add shearing drop red mushrooms
+		return $drops;
```
Comment (thebigsmileXD) [inline src/pocketmine/entity/Mooshroom.php]: I'll do that.. But no idea how to spawn a cow instead xD

File: src/pocketmine/entity/Ozelot.php [hunk 4]
```diff
@@ -25,48 +25,38 @@
...
-	const NETWORK_ID = 98;
-	public $width = 0.6;
-	public $length = 1.8;
-	public $height = 1;
- public static $range = 10;
+	const NETWORK_ID = 22;
+
+	public $width = 0.312;
+	public $length = 2.188;
+	public $height = 0.75;
```
Comment (thebigsmileXD) [inline src/pocketmine/entity/Ozelot.php]: Is that Ocelot or cat value? Seems a bit too small to me

File: src/pocketmine/entity/SnowGolem.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+        $player->dataPacket($pk);
+        parent::spawnTo($player);
+    }
+
+    public function getDrops(){
+        return [
+            drp::get(drp::SNOWBALL, 0, mt_rand(0, 15)),
+        ];
+    }
```
Comment (thebigsmileXD) [inline src/pocketmine/entity/SnowGolem.php]: // TODO: add remove pumpkin with shears

File: src/pocketmine/entity/WitherSkeleton.php [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        $player->dataPacket($pk);
+        parent::spawnTo($player);
+    }
+
+    public function getDrops(){
+        $drops = [];
+        if($this->lastDamageCause instanceof EntityDamageByEntityEvent and $this->lastDamageCause->getEntity() instanceof Player){
+            $drops = [
+                drp::get(drp::COAL, 0, mt_rand(0, 1)),
+                drp::get(drp::BONE, 0, mt_rand(0, 2))
```
Comment (thebigsmileXD) [inline src/pocketmine/entity/WitherSkeleton.php]: Stone sword?

File: src/pocketmine/entity/SnowGolem.php (hunk 5 above)
Comment (DRedDogPE) [inline src/pocketmine/entity/SnowGolem.php]: is that a thing?

File: src/pocketmine/entity/Ozelot.php (hunk 4 above)
Comment (matcracker) [inline src/pocketmine/entity/Ozelot.php]: Ocelot values, I want to improve Tameble first

File: src/pocketmine/entity/SnowGolem.php (hunk 5 above)
Comment (matcracker) [inline src/pocketmine/entity/SnowGolem.php]: True @thebigsmileXD

File: src/pocketmine/entity/Ozelot.php (hunk 4 above)
Comment (thebigsmileXD) [inline src/pocketmine/entity/Ozelot.php]: I see :)

File: src/pocketmine/entity/SnowGolem.php (hunk 5 above)
Comment (thebigsmileXD) [inline src/pocketmine/entity/SnowGolem.php]: Jup, it works in singleplayer.

File: src/pocketmine/entity/WitherSkeleton.php (hunk 6 above)
Comment (matcracker) [inline src/pocketmine/entity/WitherSkeleton.php]: I don't add item in hand on mobs

Comment (thebigsmileXD) [inline src/pocketmine/entity/WitherSkeleton.php]: Ah. Should i add them? And do they spawn with items in hand?

Comment (matcracker) [inline src/pocketmine/entity/WitherSkeleton.php]: On Enderman, Wither skeleton, Zombies, Skeleton, Pigman must have "implements InventoryHolder"

File: src/pocketmine/block/Pumpkin.php (hunk 1 above)
Comment (matcracker) [inline src/pocketmine/block/Pumpkin.php]: Mhm, true but with Irongolem it isn't correct

File: src/pocketmine/entity/Mooshroom.php (hunk 3 above)
Comment (matcracker) [inline src/pocketmine/entity/Mooshroom.php]: First we should improve the code with events

File: src/pocketmine/entity/Bat.php (hunk 2 above)
Comment (matcracker) [inline src/pocketmine/entity/Bat.php]: :D

**Comment by thebigsmileXD** [comment]
minecraft_ windows 10 edition beta 06 01 2016 21_49_22
That are almost 200 entities. 90% load, but no lag. 2 players where online :D :+1: Good job
Next thing to fix: mob server/client offset

**Comment by matcracker** [comment]
Lol, and they are only spawned, think if they have AI D:

**Comment by thebigsmileXD** [comment]
But before you had 10 entities and 100% load, and you weren't able to move.

**Pull request merged**